Fix #6634

tmilloff opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**What kind of change does this PR introduce?**
Fixes [Issue #6634](https://github.com/webpack/webpack/issues/6634)

<!-- E.g. a bugfix, feature, refactoring, build related change, etc… -->

**Did you add tests for your changes?**
No. I could use some guidance determining where in the test directory the new test should go, and what would constitute a sufficient test.

<!-- Note that we won't merge your changes if you don't add tests -->

**If relevant, link to documentation update:**
N/A
<!-- Link PR from webpack/webpack.js.org here, or N/A -->

**Summary**
Fixes bug when using an array as `config.output.library`

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
No

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**Other information**
None

sokra commented on lib/WebpackOptionsDefaulter.js:
```diff
@@ -128,6 +128,8 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 		});
 		this.set("output.devtoolNamespace", "make", options => {
+			if (Array.isArray(options.output.library))
+				return [].concat(options.output.library).join(".");
```
`options.output.library.join(".")`

tmilloff commented:
I've added the requested change in `lib/WebpackOptionsDefaulter.js`, and added an integration test in `test/configCases/source-map/array-as-output-library`

sokra commented:
Thanks
